controller/src/main/java/org/jboss/as/controller/extension/SubsystemInformation.java

```diff
@@ -56,4 +58,25 @@
      * @return the micro interface version, or {@code null} if the subsystem does not have a versioned interface
      */
     Integer getManagementInterfaceMicroVersion();
+
+
+    /**
+     * Gets full version of the subsystem
+     * @return full {@link ModelVersion}
+     */
+    default ModelVersion getManagementInterfaceVersion(){
+        return ModelVersion.create(getManagementInterfaceMajorVersion() != null ? getManagementInterfaceMajorVersion() : 0,
+                getManagementInterfaceMinorVersion() != null ? getManagementInterfaceMinorVersion() : 0,
+                getManagementInterfaceMicroVersion() != null ? getManagementInterfaceMicroVersion() : 0
+        );
+    }
+
+
+    default boolean isDeprecated(){
+        return false;
+    }
```

Comment:
This one is tricky, as we have setters as we have "two" phase subsystem initializaton.
to remove setters we would need to set everything in constructor. 
But as getSubsystemInfo() is called by initialize() and initializeParsers() 
problem is that only initialize() has full data needed to construct it without setters.
In most cases parsers are initialized first which create empty SubystemInfoImpl and than when initialize() is called it gets that info and fills it with all extra data.
To properly fix this we would need to do some bigger refactoring which is out of scope of this PR. 
I can look into it afterwards but lets keep it out of this PR for now.